artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/util/ThreadLocalByteBufferPool.java

```diff
@@ -44,6 +44,7 @@ public ByteBuffer borrow(final int size, boolean zeroed) {
       } else {
          bytesPool.set(null);
          if (zeroed) {
+            byteBuffer.clear();
```

Comment:
An alternative could be moving this before the if, and replacing the call present after the if already..but its probably safer just to leave both for all it actually does (sets a few ints).